l2geth/internal/ethapi/api.go

```diff
@@ -1358,13 +1358,11 @@ func newRPCTransaction(tx *types.Transaction, blockHash common.Hash, blockNumber
 		if meta.L1BlockNumber != nil {
 			result.L1BlockNumber = (*hexutil.Big)(meta.L1BlockNumber)
 		}
-		if meta.QueueOrigin != nil {
-			switch meta.QueueOrigin.Uint64() {
-			case uint64(types.QueueOriginSequencer):
-				result.QueueOrigin = "sequencer"
-			case uint64(types.QueueOriginL1ToL2):
-				result.QueueOrigin = "l1"
-			}
+		switch meta.QueueOrigin {
```

Comment:
I wouldn't be opposed to making `types.QueueOrigin` implement `Stringer` like so:

Where the type alias is created:
```golang
func (q *QueueOrigin) String() {
  switch q:
  case QueueOriginSequencer:
    return "sequencer"
  case QueueOriginL1ToL2:
    return "l1"
  default:
    return ""
}
```

Then here we can just do:

```
result.QueueOrigin = meta.QueueOrigin.String()
```